### PR TITLE
test(NODE-3698): sync spec tests for crud operations with `let` option

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

Support for the `let` option was added in: https://github.com/mongodb/node-mongodb-native/pull/2829

This PR pulls in spec changes from the following PRs:

- https://github.com/mongodb/specifications/pull/1105
- https://github.com/mongodb/specifications/pull/1094
- https://github.com/mongodb/specifications/pull/1150

##### Is there new documentation needed for these changes?

No

### Double check the following

- [x] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/spec/crud/unified/findOneAndUpdate-let-yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+description: "findOneAndUpdate-let"
```
Comment (jyemin) [inline test/spec/crud/unified/findOneAndUpdate-let-yml:1]: Looks like the file name got mangled.  The [original](https://github.com/mongodb/specifications/blob/master/source/crud/tests/unified/findOneAndUpdate-let.yml) ends with `.yml` not `-yml`.

Comment (baileympearson) [reply test/spec/crud/unified/findOneAndUpdate-let-yml]: Yup, good catch!

**Review by dariakp — COMMENTED** [review]
@baileympearson Did you double check that our unified test runner definitely and correctly parses the specified let option input and expectations? (I know we made the runner adjustments in the original PR, but I am a bit paranoid about these things after all the misses we've discovered after the fact, so always worth a second look)

**Comment by baileympearson** [comment]
> Did you double check that our unified test runner definitely and correctly parses the specified let option input and expectations? (I know we made the runner adjustments in the original PR, but I am a bit paranoid about these things after all the misses we've discovered after the fact, so always worth a second look)

@dariakp yes, I've confirmed that the `let` option is being parsed as expected

File: test/spec/crud/unified/findOneAndUpdate-let.json [hunk 2]
```diff
@@ -0,0 +1,217 @@
+{
+    "description": "findOneAndUpdate-let",
```
Comment (dariakp) [inline test/spec/crud/unified/findOneAndUpdate-let.json:2]: whitespace

File: test/spec/crud/unified/updateMany-let.json [hunk 3]
```diff
@@ -0,0 +1,243 @@
+{
+    "description": "updateMany-let",
```
Comment (dariakp) [inline test/spec/crud/unified/updateMany-let.json:2]: whitespace

File: test/spec/crud/unified/updateOne-let.json [hunk 4]
```diff
@@ -0,0 +1,215 @@
+{
+    "description": "updateOne-let",
```
Comment (dariakp) [inline test/spec/crud/unified/updateOne-let.json:2]: whitespace

**Review by dariakp — CHANGES_REQUESTED** [review]
Just a small issue with whitespaces, otherwise LGTM

**Pull request merged**